src/osw/core.py

```diff
@@ -636,6 +742,24 @@ def __init__(self, **data):
                 self.parallel = True
             if self.parallel is None:
                 self.parallel = False
+            if self.overwrite is None:
+                self.overwrite = self.__fields__["overwrite"].default
+            self._overwrite_per_class = {"by name": {}, "by type": {}}
+            if self.overwrite_per_class is not None:
+                for param in self.overwrite_per_class:
+                    model_name = param.model.__name__
+                    model_type = param.model.__fields__["type"].default[0]
+                    if (
+                        model_name in self._overwrite_per_class["by name"].keys()
+                        or model_type in self._overwrite_per_class["by type"].keys()
+                    ):
+                        raise ValueError(
+                            f"More than one OverwriteClassParam for the class "
+                            f"'{model_type}' ({model_name}) has been passed in the "
+                            f"list to 'overwrite_per_class'!"
+                        )
+                    self._overwrite_per_class["by name"][model_name] = param
+                    self._overwrite_per_class["by type"][model_type] = param
 
     def store_entity(
```

Comment:
Can we separate the application of the override policy in another (util) function?
e.g. apply_override_policy(org_page, new_page, policy)
This would improve clearness & testability of the code

Reply:
I'm not sure I understood correctly.

Should this util function be a wrapper for the store_entity function and a method of osw.core.OSW? 
Or should StoreEntityParam define another method that is called by the __init__ or during execution of store_entity (when the overwrite_per_class property is None)

Reply:
A sketch of what could be done:
We could separate the application of the overwrite of a page (locally setting the slot content + calling page.edit()) into a separate function osw.core.OSW.apply_overwrite_policy(org_page, entity, policy) -> osw.core.wtsite.WtPage
- a params object would be defined with a validator checking that org_page and entity have the same uuid (for cases where apply_overwrite_policy is called independently of store_entity)
- the store_entity funtion would still call get_page to get the original page (if exists)
- therefore the set_content_and_edit_page method would need to be moved to e.g. wtsite.WtPage
If it is not desired to access apply_overwrite_policy independently, we could also define it as a function inside store_entity, as done for set_content_and_edit_page